components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceBasedSSHKeyVerifier.java

```diff
@@ -0,0 +1,211 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.ssh;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.charset.Charset;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.util.ResourceHelper;
+import org.apache.sshd.ClientSession;
+import org.apache.sshd.client.ServerKeyVerifier;
+import org.bouncycastle.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ServerKeyVerifier that takes a camel resource as input file to validate the server key against.
+ *
+ */
+public class ResourceBasedSSHKeyVerifier implements ServerKeyVerifier {
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+
+    private CamelContext camelContext;
+    private boolean failOnUnknownHost;
+    private String knownHostsResource;
+
+    public ResourceBasedSSHKeyVerifier(CamelContext camelContext, String knownHostsResource) {
+        this.camelContext = camelContext;

```

Comment:
Hi,

thanks for you suggestions.

I wanted to parse the file on every connection attempt (this is the same behavior as other ssh clients). Therefore I can't parse it beforehand. 

Another approach would be to resolve it to an URL by the ResourceHelper class - However I am not sure if this is possible for every URI. Would you prefer this approach?

Sascha